spec/statement.dd

```diff
@@ -1163,14 +1163,31 @@ switch (name)
         less error prone. char, wchar and dchar strings are allowed.
         )
 
-        $(P $(D Implementation Note:) The compiler's code generator may
+        $(P If a $(GLINK CaseStatement), $(GLINK CaseRangeStatement), or $(GLINK DefaultStatement)
+        appears in the top level of the $(GLINK ScopeStatement), then a new scope for local
+        symbols is introduced immediately following it, extending to the first appearance of one of
+        $(GLINK CaseStatement), $(GLINK CaseRangeStatement), $(GLINK DefaultStatement), `}`,
+        the end of a $(GLINK MixinStatement), or the end of the file.
+        )
+
+        $(IMPLEMENTATION_DEFINED The implementation may
         assume that the case
         statements are sorted by frequency of use, with the most frequent
         appearing first and the least frequent last. Although this is
         irrelevant as far as program correctness is concerned, it is of
         performance interest.
         )
 
+        $(BEST_PRACTICE
+        $(OL
+        $(LI Use a $(GLINK ScopeBlockStatement) rather than a $(GLINK ScopeStatement)
+        for the $(B switch) body. (I.e. use `{ }`.))
+        $(LI $(GLINK CaseStatement), $(GLINK CaseRangeStatement), and $(GLINK DefaultStatement)
+        should appear directly in the `{ }` of the $(GLINK ScopeStatement), rather than
+        in nested `{ }`.)
+        )
+        )
```

Comment:
Is there an example where allowing CaseStatement and the others in nested scopes is actually useful?

Reply:
I wouldn't allow it in our code, but people do use it in C. In any case, I'm trying to document what it does, not what we'd like it to do.